src/tools/rados/rados.cc

```diff
@@ -431,15 +431,15 @@ static int do_put(IoCtx& io_ctx, RadosStriper& striper,
     }
     indata.append(buffer::ptr(buffer::create_static(count, buf)));
     if (use_striper) {
-      if (offset == 0)
+      if (obj_offset == 0)
```

Comment:
if the obj_offset is not 0, we should be able to punch a hole using
```
$ rados put obj-name path/to/a/zero-sized-file 128
```

but we just stop if the `count` is 0 with your change, and a zero-sized object is created instead.